Add custom styling for CLI output utils

strickvl opened the pull request:
## Describe changes
I added an optional argument for CLI util functions which allows for the adding of custom styles.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

schustmi commented:
I thought our `cli_utils.warning` etc are sort of a way to have uniform messages for logs of a certain importance. In case we merge this, why would I ever call `cli_utils.warning(..., custom_style)` instead of `console.print(..., custom_style)`?

strickvl commented:
@schustmi You're completely right, though I see the `cli_utils` as also having some kind of semantic or signalling function for us internally. I came across an example in a separate PR where I wanted to output a 'warning' type of message, but I wanted it to be italicised, so thought of making a more general change where people could preserve that fact that a message is a 'warning' while allowing for some more specific customisation as needed. That at least was my thinking behind this PR.

strickvl commented:
@schustmi I addressed your comments. I limited what people can do, and allowed the changes only for `warning` and `declare` functions (since it doesn't make sense to allow it for `title`).

schustmi commented on src/zenml/console.py:
```diff
@@ -16,13 +16,13 @@
+    "error": "red",
+}
+
+zenml_custom_theme = Theme(zenml_style_defaults)
```
Is this theme something that `rich` uses internally as well, or can we remove it as it's now not used by our `cli_utils.declare/warning` functions?
And if we can't, should we somehow read the colors for the `cli_utils` from here in order to have just one place where we define them?

strickvl replied:
The problem is that when someone imports our `console`, they'll want to get these presets built-in, so that's why we need to keep it.

Also, the other problem is that these aren't just colours, so it's not just a simple matter of parsing the k:v pairs. The alternative would be to store these as `Style` objects, which might make sense?

schustmi replied:
Can you have `Style` objects in a rich `Theme`? If yes I feel like that would be a good compromise, and in `cli_utils.warning` we could get that style from the dict and modify it. But if not possible or too much work, feel free to ignore.

strickvl replied:
Themes are basically mappings of names of styles with their corresponding `Style` objects. I've made changes accordingly.

schustmi commented on src/zenml/cli/utils.py:
```diff
@@ -78,13 +78,23 @@ def confirmation(text: str, *args: Any, **kwargs: Any) -> bool:
+        italic: Optional boolean to italicize the text.
     """
-    console.print(text, style="info")
+    style = Style(color="cyan", dim=True)
+    if bold:
+        style.bold = True
+    if italic:
+        style.italic = True
```
```suggestion
    style = Style(color="cyan", dim=True, bold=bold, italic=italic)
```

schustmi commented on src/zenml/cli/utils.py:
```diff
@@ -100,16 +110,21 @@ def error(text: str) -> None:
+        italic: Optional boolean to italicize the text.
     """
-    console.print(text, style=custom_style)
+    style = Style(color="yellow")
```
```suggestion
    style = Style(color="yellow", dim=True, bold=bold, italic=italic)
```

AlexejPenner reviewed (approved):
LGTM !!! - lets style it up
